rspace/src/main/scala/coop/rchain/rspace/IStore.scala

```diff
@@ -86,40 +82,32 @@ trait IStore[C, P, A, K] {
 
   def withTrieTxn[R](txn: Transaction)(f: TrieTransaction => R): R
 
-  protected val _trieUpdates: SyncVar[Seq[TrieUpdate[C, P, A, K]]] =
-    SyncVarOps.create(Seq.empty)
+  private val _trieUpdates: AtomicAny[(Long, List[TrieUpdate[C, P, A, K]])] =
+    AtomicAny[(Long, List[TrieUpdate[C, P, A, K]])]((0L, Nil))
 
-  def trieDelete(key: Blake2b256Hash, gnat: GNAT[C, P, A, K]) = {
-    val count   = _trieUpdateCount.getAndIncrement()
-    val currLog = _trieUpdates.take()
-    _trieUpdates.put(currLog :+ TrieUpdate(count, Delete, key, gnat))
-  }
+  def trieDelete(key: Blake2b256Hash, gnat: GNAT[C, P, A, K]): Unit =
+    _trieUpdates.getAndTransform((t: (Long, List[TrieUpdate[C, P, A, K]])) => {
+      (t._1 + 1, TrieUpdate(t._1, Delete, key, gnat) :: t._2)
```

Comment:
Why not do `case (count, updates) =>`? Does unapply affect performance that much?